### PR TITLE
Update sshd-core to 1.3.0

### PR DESCRIPTION
**Pull request by MRamonLeon**
Update sshd-core library to avoid a dependency vulnerability. It's not exploitable, but it shows up on every security scan.

### CODE REVIEW

**Comment by jglick** [comment]
Plugin needs a maintainer; there are a bunch of open PRs that are conflicting and need various degrees of cleanup and manual testing.

**Comment by MRamonLeon** [comment]
> Plugin needs a maintainer; there are a bunch of open PRs that are conflicting and need various degrees of cleanup and manual testing.

WDYM @jglick? We need to manually test or run PCT?

**Comment by jglick** [comment]
> We need to manually test or run PCT?

Manually test using an actual `*.hpi` since this uses `pluginFirstClassLoader` so `JenkinsRule` would not exercise the nondefault modes realistically even if you figured out how to use `docker-fixtures` to simulate an agent lacking an `ssh-agent` executable; or finally rip out these modes which are hard to support and just leave in the simple native executable mode.

File: src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java [hunk 1]
```diff
@@ -6,15 +6,17 @@
 import java.io.OutputStream;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.logging.Logger;
+
+import org.apache.sshd.server.ServerFactoryManager;
```
Comment (alecharp) [inline src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java:10]: I don't see a usage for this import.

File: src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java [hunk 2]
```diff
@@ -6,17 +6,17 @@
...
 import org.apache.sshd.server.Command;
 import org.apache.sshd.server.CommandFactory;
 import org.apache.sshd.server.Environment;
 import org.apache.sshd.server.ExitCallback;
 import org.apache.sshd.server.auth.pubkey.PublickeyAuthenticator;
-import org.apache.sshd.server.command.UnknownCommand;
 import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.apache.sshd.server.scp.UnknownCommand;
 import org.apache.sshd.server.session.ServerSession;
-import org.apache.sshd.server.session.SessionFactory;
```
Comment (MRamonLeon) [inline src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java:19]: Not used

**Comment by MRamonLeon** [comment]
I've run the ATH Test: https://github.com/jenkinsci/acceptance-test-harness/blob/14b831e4a5ff43f8eb03fe1029779785e510049f/src/test/java/plugins/WorkflowPluginTest.java#L216-L263 successfully with the new version. Command line:

ssh-agent.hpi is this version.

```
export LOCAL_JARS=ssh-agent.hpi; run.sh firefox latest -DforkCount=1 -Dmaven.test.failure.ignore=true -B -Dtest=WorkflowPluginTest#sshGitInsideDocker
```

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running plugins.WorkflowPluginTest
...
INFO: Plugin ssh-agent@1.20-SNAPSHOT (private-5fc21a1f-rleon) is installed
...
INFO: Jenkins is stopped
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 123.33 s - in plugins.WorkflowPluginTest
2020-06-26 15:04:44,877 [LittleProxy-JVM-shutdown-hook] INFO   (DefaultHttpProxyServer.java:436) org.littleshoot.proxy.impl.DefaultHttpProxyServer - Shutting down proxy server immediately (non-graceful)
2020-06-26 15:04:44,878 [LittleProxy-JVM-shutdown-hook] INFO   (DefaultHttpProxyServer.java:469) org.littleshoot.proxy.impl.DefaultHttpProxyServer - Closing all channels (non-graceful)
2020-06-26 15:04:44,883 [LittleProxy-JVM-shutdown-hook] INFO   (ServerGroup.java:212) org.littleshoot.proxy.impl.ServerGroup - Shutting down server group event loops (non-graceful)
2020-06-26 15:04:44,885 [LittleProxy-JVM-shutdown-hook] INFO   (DefaultHttpProxyServer.java:450) org.littleshoot.proxy.impl.DefaultHttpProxyServer - Done shutting down proxy server
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:15 min
[INFO] Finished at: 2020-06-26T15:04:45Z
[INFO] ------------------------------------------------------------------------

```

**Pull request merged**